src/semgrep_agent/targets.py

```diff
@@ -208,17 +214,58 @@ def _get_target_files(self) -> List[Path]:
 
         return relative_paths
 
-    def _abort_if_dirty(self) -> None:
+    @_dirty_paths_by_status.default
+    def get_dirty_paths_by_status(self) -> Dict[str, List[Path]]:
```

Comment:
```suggestion
    def get_dirty_paths_by_status(self) -> Mapping[str, Sequence[Path]]:
```

Reply:
@nbrahms aren't the more specific ones better? With the earlier annotation, dict methods would be suggested on the attribute as well.